### PR TITLE
Add 'warn' alternative to 'validate'

### PR DESCRIPTION
**Pull request by timhwang21**
Referencing #1779 and #1649.

To use, add a function warn() to reduxForm decorator. This function should
be similar to validate(). Behavior will be identical, except warnings do not
mark a form as invalid.

(If I missed any steps in the PR process, please let me know. Thanks!)

Also updated tests to cover new feature. There are two odd tests marked with
TODO in reduxForm.spec.js involving unexpected renders, though.

### CODE REVIEW

File: src/__tests__/Field.spec.js [hunk 1]
```diff
@@ -1035,6 +1078,101 @@ const describeField = (name, structure, combineReducers, expect) => {
...
+      // username field has warning
+      expect(usernameInput.calls[ 0 ].arguments[ 0 ].meta.warning).toBe('Recommended')
+
+      // update username field so it passes
+      usernameInput.calls[ 0 ].arguments[ 0 ].input.onChange('erikras')
+
+      // username input rerendered twice, once for value, once for sync warning
+      expect(usernameInput.calls.length).toBe(3)
+
+      // should be valid now
```
Comment (timhwang21) [inline src/__tests__/Field.spec.js]: just realized some of these comments are nonsensical now. I *could* re-add the validity checks (warning should not invalidate form), or just take out the comments.

**Comment by erikras** [comment]
Excellent work. Can you do the same process in [the docs](https://github.com/erikras/redux-form/tree/master/docs)?

**Comment by timhwang21** [comment]
Thanks for the response!

* Described warnings usage in `reduxForm.md`
* Added simple warning to Sync Validation example -- if this is unnecessary, can be removed.

Let me know if you need anything else. Thanks again!

**Comment by erikras** [comment]
Needs to be in `Field.md`, too.

**Comment by timhwang21** [comment]
Done!

**Comment by erikras** [comment]
😞  Sorry, your PR was first, but something got merged that is causing a conflict. Can you resolve? I will merge asap.

**Comment by timhwang21** [comment]
Yep, no problem -- fixed it and pushed.

(Would it be better practice to squash my four commits and add the issue number? Thanks!)

**Comment by elodszopos** [comment]
I'm actually looking forward to this one too, always wondered why it was always black or white only. 👍

**Comment by timhwang21** [comment]
Thanks! 😄

**Comment by ilxanlar** [comment]
Finally got what I was waiting for, Thank you guys.

**Comment by btroop** [comment]
thanks @timhwang21 !

**Comment by erikras** [comment]
Published in [`v6.1.0`](https://github.com/erikras/redux-form/releases/tag/v6.1.0).

**Pull request merged**